Modules/Forum/interfaces/interface.ilForumNotificationMailData.php

```diff
@@ -10,112 +10,112 @@ interface ilForumNotificationMailData
     /**
      * @return int
      */
-    public function getRefId();
+    public function getRefId() : int;
 
     /**
      * @return int
      */
-    public function getObjId();
+    public function getObjId() : int;
 
     /**
      * @return int frm_data.top_pk
      */
-    public function getForumId();
-    
+    public function getForumId() : int;
+
     /**
      * @return string frm_data.top_name
      */
-    public function getForumTitle();
+    public function getForumTitle() : string;
 
     /**
      * @return int
      */
-    public function getThreadId();
-    
+    public function getThreadId() : int;
+
     /**
      * @return string frm_threads.thr_subject
      */
-    public function getThreadTitle();
+    public function getThreadTitle() : string;
 
     /**
      * @return int
      */
-    public function getPostId();
+    public function getPostId() : int;
 
     /**
      * @return string frm_posts.pos_subject
      */
-    public function getPostTitle();
+    public function getPostTitle() : string;
 
     /**
      * @return string frm_posts.pos_message
      */
     public function getPostMessage();
-    
+
     /**
-     * @return string frm_posts.pos_author_id
+     * @return int frm_posts.pos_author_id
      */
     public function getPosAuthorId();
 
     /**
      * @return int
      */
-    public function getPostUpdateUserId();
-    
+    public function getPostUpdateUserId() : int;
+
     /**
-     * @return string frm_posts.pos_display_user_id
+     * @return int frm_posts.pos_display_user_id
      */
-    public function getPosDisplayUserId();
+    public function getPosDisplayUserId() : int;
 
     /**
      * @return string frm_posts.pos_usr_alias
      */
-    public function getPosUserAlias();
+    public function getPosUserAlias() : string;
 
     /**
-     * @param \ilLanguage $user_lang
+     * @param ilLanguage $user_lang
      * @return string
      */
-    public function getPostUserName(\ilLanguage $user_lang);
+    public function getPostUserName(ilLanguage $user_lang) : string;
 
     /**
      * @return string frm_posts.pos_date
      */
-    public function getPostDate();
+    public function getPostDate() : string;
 
     /**
-     * @return string frm_posts.pos_update
+     * @return string|null frm_posts.pos_update
      */
-    public function getPostUpdate();
+    public function getPostUpdate() : string|null;
 
     /**
-     * @param \ilLanguage $user_lang
-     * @return string
+     * @param ilLanguage $user_lang
+     * @return string|null
      */
-    public function getPostUpdateUserName(\ilLanguage $user_lang);
-    
+    public function getPostUpdateUserName(ilLanguage $user_lang) : string|null;
```

Comment:
We MUST NOT use `Union Types`. The code MUST be compatible with PHP 7.4.

```suggestion
    public function getPostUpdateUserName(ilLanguage $user_lang) : ?string;
```